dali/operators/coord/coord_transform.h

```diff
@@ -0,0 +1,307 @@
+    for (int i = 0; i < N; i++) {
+      // ...and replace the number of vector components
+      output_descs[0].shape.tensor_shape_span(i).back() = output_pt_dim_;
+    }
+    return true;
+  }
+
+  void RunImpl(workspace_t<Backend> &ws) override {
+    auto &in = ws.template InputRef<Backend>(0);
+    DALIDataType in_type = in.type().id();
+    VALUE_SWITCH(output_pt_dim_, static_out_dim, COORD_TRANSFORM_DIMS, (
+        VALUE_SWITCH(input_pt_dim_, static_in_dim, COORD_TRANSFORM_DIMS, (
+            TYPE_SWITCH(in_type, type2id, InputType, COORD_TRANSFORM_INPUT_TYPES, (
+                if (dtype_ == in_type) {
+                  RunTyped<InputType, InputType, static_out_dim, static_in_dim>(ws);
+                } else {
+                  DALI_ENFORCE(dtype_ == DALI_FLOAT, make_string("CoordTransform output type "
+                    "must be the same as input type or `float`. Got: ", dtype_));
+                  RunTyped<float, InputType, static_out_dim, static_in_dim>(ws);
+                }
+              ), (  // NOLINT
+                DALI_FAIL(make_string("Unsupported input type: ", in_type));
+              )  // NOLINT
+            )
+          ), (  // NOLINT
+            DALI_FAIL(make_string("Unsupported input point dimensionality: ", input_pt_dim_));
+          )   // NOLINT
+        )  // NOLINT
+      ), (  // NOLINT
+        DALI_FAIL(make_string("Unsupported input point dimensionality: ", input_pt_dim_));
+      )  // NOLINT
+    )  // NOLINT
+  }
+
+  void PrepareTransformArguments(const workspace_t<Backend> &ws,
+                                 const TensorListShape<> &input_shape) {
+    input_pt_dim_ = 0;
+    output_pt_dim_ = 0;
+
+    DALI_ENFORCE(input_shape.sample_dim() >= 2,
+      "CoordTransform expects an input with at least 2 dimensions.");
+
+    const int N = input_shape.num_samples();
+    input_pt_dim_ = 0;
+    for (int i = 0; i < N; i++) {
+      auto sample_shape = input_shape.tensor_shape_span(i);
+      if (volume(sample_shape) == 0)
+        continue;
+      int pt_dim = input_shape.tensor_shape_span(i).back();
+      if (input_pt_dim_ == 0) {
+        input_pt_dim_ = pt_dim;
+      } else {
+        DALI_ENFORCE(pt_dim == input_pt_dim_, make_string("The point dimensions must be the same "
+        "for all input samples. Got: ", input_shape, "."));
+      }
+    }
+    if (input_pt_dim_ == 0)
+      return;  // data is degenerate - empty batch or a batch of empty tensors
+
+    if (HasFusedMT()) {
+      ProcessMatrixArg(ws, "MT", N);
+    } else {
+      ProcessMatrixArg(ws, "M", N);
+      ProcessTranslationArg(ws, "T", N);
+    }
+  }
+
+  bool HasFusedMT() const {
+    return has_mt_ || has_mt_input_;
+  }
+
+  bool HasMatrixRegularArg() const {
+    return has_matrix_ || has_mt_;
+  }
+
+  bool HasMatrixInputArg() const {
+    return has_matrix_input_ || has_mt_input_;
+  }
+
+  void ProcessMatrixArg(const workspace_t<Backend> &ws, const char *name, int N) {
+    bool is_fused = HasFusedMT();
```

Comment:
I don't like this. `ProcessMatrixArg` seems to be a "generic" function, because it can process "M" or "MT", which you specify by name. However, you are checking explicitly `HasFusedMT()` to determine whether this is MT or M.

How about removing the argument `name` and let this function figure it out?
Another option would be to determine whether it is MT by checking the `name`.

My point is that there should be a single source of truth.